sunxspex/sunxspex_fitting/likelihoods.py

```diff
@@ -250,10 +254,18 @@ def cstat_loglikelihood(self, model_counts, observed_counts, observed_count_erro
         A float, the c-stat log-likelihood (to be maximised).
         """
 
+        model_counts, observed_counts, observed_count_errors = np.array(model_counts).ravel(), np.array(observed_counts), np.array(observed_count_errors)
+
         # C-stat (XSPEC) - has data term in it so the lower the -2*ln(L) number the better the fit, this is the ln(L) though
         # Best value is 0, if obs_counts>mod_counts or obs_counts<mod_counts then likelihoods<0
         # Obvious since e^0=1
-        likelihoods = np.array(observed_counts) * (np.log(np.array(model_counts)/np.array(observed_counts)) + 1) - np.array(model_counts)
+        likelihoods = observed_counts * (np.log(model_counts/observed_counts) + 1) - model_counts
+
+        # if zero observed counts in a bin, then Stirling Approx. not good (only good for obs_counts>=1, if =0 then ->-inf, but Poisson->-mod_counts)
+        # defer to Poisson for the nans in this case
+        _zero_data = (observed_counts == 0) & (model_counts > 0)
+        if len(observed_counts[_zero_data]) > 0:
+            likelihoods[_zero_data] = -model_counts[_zero_data]
```

Comment:
Is the if statement necessary? `_zero_data` is a boolean mask so should always work. Unless it a perforce thing but might be faster too?